config/config_helpers.go

```diff
@@ -301,6 +307,93 @@ func getAWSAccountID(include *IncludeConfig, terragruntOptions *options.Terragru
 	return *identity.Account, nil
 }
 
+// Return the output from the state of another module, managed by terragrunt. This function will parse the provided
+// terragrunt config and extract the desired output from the remote state. Note that this will error if the targetted
+// module hasn't been applied yet. If you omit the output name, this will return a map of all the outputs from the
+// module.
+func getTerragruntOutput(params []string, include *IncludeConfig, terragruntOptions *options.TerragruntOptions) (*cty.Value, error) {
+	// args check: make sure the expected args are passed
+	numParams := len(params)
+	if numParams == 0 {
+		return nil, errors.WithStackTrace(EmptyStringNotAllowed("the target terragrunt config parameter"))
+	}
+	if numParams > 2 {
+		return nil, errors.WithStackTrace(WrongNumberOfParams{Func: "get_output", Expected: "1 or 2", Actual: numParams})
+	}
+
+	// target config check: make sure the target config exists
+	targetConfig := params[0]
+	if util.IsDir(targetConfig) {
+		targetConfig = util.JoinPath(targetConfig, DefaultTerragruntConfigPath)
```

Comment:
TODO for the future: we could support reading outputs from normal Terraform modules (i.e., those not meant to be used with Terragrunt) by running `terraform output` instead of `terragrunt output`. Not particularly high priority for now though!